### PR TITLE
chore: add requestAsBrowser migration to docs

### PR DESCRIPTION
**Pull request by szmarczak**

### CODE REVIEW

File: docs/upgrading/upgrading_v3.md [hunk 1]
```diff
@@ -233,6 +233,162 @@ const crawler = new BasicCrawler({
 });
 ```
 
+### How does `sendRequest` work?
+
+The `sendRequest` function looks like this:
```
Comment (B4nan) [inline docs/upgrading/upgrading_v3.md:238]: we should have this documented elsewhere too, maybe a separate guide mentioning how to work with gotScraping, `sendRequest` helper would be one section there. then the upgrading guide could just link it, right now it feels like this change takes quite a substantial space on the upgrading guide :]

File: packages/basic-crawler/src/internals/basic-crawler.ts [hunk 2]
```diff
@@ -785,7 +786,7 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
                     requestQueue: await this.getRequestQueue(),
                 });
             },
-            sendRequest: async (overrideOptions?: Partial<GotOptionsInit>) => {
+            sendRequest: async (overrideOptions?: GotOptionsInit) => {
```
[the remark this answers is not in the document]
Comment (szmarczak) [reply packages/basic-crawler/src/internals/basic-crawler.ts:789]: Yes. https://github.com/sindresorhus/got/blob/09e8273931e4ccd4d87f4b0b315a2b063e9083e7/source/core/options.ts#L655-L660

File: docs/upgrading/upgrading_v3.md [hunk 3]
```diff
@@ -233,6 +233,162 @@ const crawler = new BasicCrawler({
...
+    headerGeneratorOptions: {
+        devices: ['mobile'], // or ['desktop']
+        locales: ['en-US'],
+    },
+});
+```
+
+#### `timeoutSecs`
+
+In order to set a timeout, use `timeout.request` (which is milliseconds now !!!).
```
[the remark this answers is not in the document]
Comment (szmarczak) [reply docs/upgrading/upgrading_v3.md:342]: I don't use tiktok.

File: docs/upgrading/upgrading_v3.md (hunk 1 above)
Comment (szmarczak) [reply docs/upgrading/upgrading_v3.md:238]: created an issue about this

**Pull request merged**